docs/datatypes/tset.md

```diff
@@ -0,0 +1,251 @@
+---
+id: datatypes_tset
+title: "TSet"
+---
+
+A `TSet[A]` is transactional data structure built on top of `TMap[A]`.
+
+## Create a TSet
+
+Creating an empty `TSet`:
+
+```scala mdoc:silent
+import zio._
+import zio.stm._
+
+val emptyTSet: STM[Nothing, TSet[Int]] = TSet.empty[Int]
+```
+
+Or creating a `TSet` with specified values:
+
+```scala mdoc:silent
+import zio._
+import zio.stm._
+
+val specifiedValuesTSet: STM[Nothing, TSet[Int]] = TSet.make(1, 2, 3)
+```
+
+Or creating a `TSet` from existing `Iterable` structure:
+
+```scala mdoc:silent
+import zio._
+import zio.stm._
+
+val iterableTSet: STM[Nothing, TSet[Int]] = TSet.fromIterable(List(1, 2, 3))
+```
+
+## Put an element to a `TSet`
+
+Putting new element to a `TSet` extends it if specified element is not already present:
```

Comment:
Split in two, e.g.:

```
New element can be added to the set in the following way:

<example>

In case the set already contains the element, no modification will happen.
```